File: DemoClient/projects/ng-realtime-database/src/lib/models/command/create-user-command.ts
import {CommandBase} from './command-base';

export class CreateUserCommand extends CommandBase {
  userName: string;
  email: string;
  password: string;
  roles: string[];
  additionalData: { [key: string]: any };

  constructor(userName: string, email: string, password: string, roles: string[], addtionalData: { [key: string]: any }) {
    super('CreateUserCommand');
    this.userName = userName;
    this.email = email;
    this.password = password;
    this.roles = roles;
    this.additionalData = addtionalData;
  }
}

File: RealtimeDatabase/Command/CreateUser/CreateUserCommand.cs
using System.Collections.Generic;
using Newtonsoft.Json.Linq;

namespace RealtimeDatabase.Command.CreateUser
{
    public class CreateUserCommand : CommandBase
    {
        public string UserName { get; set; }

        public string Email { get; set; }

        public string Password { get; set; }

        public string[] Roles { get; set; } = new string[0];

        public Dictionary<string, JValue> AdditionalData { get; set; } = new Dictionary<string, JValue>();
    }
}
